### PR TITLE
fix(test): Add a timeout to the non_blocking_logger test

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

The non_blocking_logger test timed out on PR #5164 and produced 3 GB of logs:
https://github.com/ZcashFoundation/zebra/actions/runs/3177218716/jobs/5177415689

If that happens again, we want to end the test sooner, so the bug is easier to diagnose.

## Review

Anyone can review this PR.

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [x] Does the code do what the ticket and PR says?
  - [x] How do you know it works? Does it have tests?

### CODE REVIEW

File: zebrad/tests/acceptance.rs [hunk 1]
```diff
@@ -1353,7 +1353,8 @@ fn non_blocking_logger() -> Result<()> {
         let zebra_rpc_address = config.rpc.listen_addr.unwrap();
 
         let dir = testdir()?.with_config(&mut config)?;
-        let mut child = dir.spawn_child(args!["start"])?;
+        let mut child = dir.spawn_child(args!["start"])?.with_timeout(TINY_CHECKPOINT_TIMEOUT);
+
```
Comment (teor2345) [inline zebrad/tests/acceptance.rs]:
```suggestion
        let mut child = dir
            .spawn_child(args!["start"])?
            .with_timeout(TINY_CHECKPOINT_TIMEOUT);

```

**Comment by teor2345** [comment]
Failed due to #5069:
https://github.com/ZcashFoundation/zebra/actions/runs/3178405970/jobs/5180280185#step:6:110

**Comment by teor2345** [comment]
@mergifyio update

**Review by dconnolly — APPROVED** [review]

**Comment by gustavovalverde** [comment]
@Mergifyio update

**Comment by gustavovalverde** [comment]
@mergifyio refresh

**Comment by teor2345** [comment]
@mpguerra this bug was actually caused by PR #5032, it is unrelated to PR #4894.

**Pull request merged**